File: src/main/java/org/sindeaux/autoconsulta/entityes/Medico.java
package org.sindeaux.autoconsulta.entityes;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.OneToMany;

import java.util.List;

@Entity
public class Medico extends Funcionario{

    @Column(name = "CRM_MEDICO")
    private String crm;

    @Column(name = "ESPECIALIZACAO")
    private String especializacao;

    @OneToMany(mappedBy = "medico", fetch = FetchType.LAZY)
    private List<Consulta> consultas;
}

File: src/main/kotlin/com/sindeaux/autoconsulta/entity/Medico.kt
package com.sindeaux.autoconsulta.entity

import Pessoa
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.OneToMany

@Entity
class Medico : Pessoa() {
    @Column(name = "CRM_MEDICO")
    open var crm: String? = null

    @Column(name = "ESPECIALIZACAO")
    open var especializcao: String? = null

    @OneToMany(mappedBy = "medico", fetch = FetchType.LAZY)
    open var consultas: List<Consulta>? = null;

}